pentasharp/Controllers/TaxiCompanyController.cs

```diff
@@ -71,17 +71,35 @@ public IActionResult EditCompany(int id, [FromBody] TaxiCompanyViewModel model)
         [HttpDelete("DeleteCompany/{id}")]
         public IActionResult DeleteCompany(int id)
         {
-            var company = _context.TaxiCompanies.Include(c => c.Taxis).FirstOrDefault(c => c.TaxiCompanyId == id);
+            var company = _context.TaxiCompanies
+                                  .Include(c => c.Taxis) 
+                                  .FirstOrDefault(c => c.TaxiCompanyId == id);
+
             if (company == null)
             {
                 return NotFound(new { success = false, message = "Company not found." });
             }
 
-            _context.TaxiCompanies.Remove(company);
+          
+            company.IsDeleted = true;
+            company.UpdatedAt = DateTime.UtcNow;
+
+       
+            foreach (var taxi in company.Taxis)
+            {
+                taxi.IsDeleted = true;
+                taxi.UpdatedAt = DateTime.UtcNow;
+            }
+

```

Comment:
reformat code, remove unnecessary extra lines such as line 86 or 87, lines 93,94,95 keep only one of them